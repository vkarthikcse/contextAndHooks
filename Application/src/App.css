.App {
    text-align: center;
    padding-top: 20px;
}

.App-logo {
    height: 40vmin;
}

.App-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: #09d3ac;
}

.Course-list {
    background-color: #ccc;
    width: 50%;
    margin: 0 auto;
    border-radius: 5px;
}

.Course-list:hover{
  cursor: pointer;
}

form div {
    padding-bottom:20px;
}

.course-name {
    padding: 10px;

}

.course-id {
    padding: 23px;
}

button {
    margin-left: 40px;
    padding: 10px;
    width: 100px;
    border-radius: 5px;
    box-shadow: none;

}
button:hover{
    background-color: black ;
    color: white;
    cursor: pointer;
}