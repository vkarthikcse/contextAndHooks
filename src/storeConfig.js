// import React, {useContext} from "react";

// export const CourseAppContext = React.createContext();
// export const CourseAppProvider = CourseAppContext.Provider;



import React, { createContext, useReducer, useContext } from "react";
import reducer, { initialState } from "./reducers/courseReducer";

const customMiddleware = store => next => action => {
  console.log("Action Triggered");
  console.log(action);
  next(action);
};

const Store = createContext();

const compose = (...funcs) => x =>
  funcs.reduceRight((composed, f) => f(composed), x);

const createStore = (state, dispatch , middlewares) => {
  if (typeof middlewares !== "undefined") {
    // return middlewares(createStore)(reducer, initialState);
    const middlewareAPI = {
      getState: () => state,
      dispatch: action => dispatch(action)
    };
    const chain = middlewares.map(middleware => middleware(middlewareAPI));
    const enhancedDispatch = compose(...chain)(dispatch);
    return { state, dispatch: enhancedDispatch };
  }

  return { state, dispatch };
};

const Provider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const store = createStore(state, dispatch, [customMiddleware]);
  return <Store.Provider value={store}>{children}</Store.Provider>;
};


const connect = (
  mapStateToProps = () => {},
  mapDispatchToProps = () => {}
) => WrappedComponent => {
  return props => {
    const { dispatch, state } = useContext(Store);
    return (
      <WrappedComponent
        dispatch={dispatch}
        {...mapStateToProps(state, props)}
        {...mapDispatchToProps(dispatch)}
      />
    );
  };
};

export { connect, Store, Provider };
