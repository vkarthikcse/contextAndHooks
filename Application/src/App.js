import React from "react";
import "./App.css";
import FormComponent from "./components/FormComponent";
import CourseList from "./components/CourseList";

function App() {
    return (
        
            <div className="App">
                <FormComponent />
                <CourseList />
            </div>
        
    );
}

export default App;
