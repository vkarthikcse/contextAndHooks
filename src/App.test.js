import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { Provider } from "./storeConfig";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {act} from "react-test-renderer";

configure({ adapter: new Adapter() });
it("renders without crashing", () => {
  let div;
    act(() => {
      
        div = mount(
            <Provider>
                <App />
            </Provider>
        );
    });

    // expect(div.props().courses).toEqual([
    //     {
    //         id: 1,
    //         name: "course1"
    //     },
    //     {
    //         id: 2,
    //         name: "course2"
    //     },
    //     {
    //         id: 3,
    //         name: "course3"
    //     }
    // ]);
});
