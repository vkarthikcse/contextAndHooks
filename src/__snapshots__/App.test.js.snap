// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="form-header"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          className="course-name"
          htmlFor="coursename"
        >
          Course name
        </label>
        <input
          id="coursename"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          className="course-id"
          htmlFor="courseid"
        >
          Course id
        </label>
        <input
          id="courseid"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </div>
      <div>
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
